app/models/concerns/wakes/metrics/google_analytics_pageviews.rb

```diff
@@ -19,24 +15,12 @@ def google_analytics_profile_id
           end
         end
 
-        def self.enqueue_pageview_count_updates(count)
-          ordered_for_analytics_worker.needs_analytics_update.limit(count).each do |location|
-            Wakes::GoogleAnalyticsPageviewJob.perform_later(location)
-          end
-        end
-
-        def self.ordered_for_analytics_worker
-          order("document->'pageview_count_checked_at' ASC NULLS FIRST")
+        def pageview_count
+          pageview_counts.sum { |_year, count| count }
         end
 
-        def self.needs_analytics_update
-          # either it's never been updated
-          # OR it's canonical and it has not been updated through yesterday
-          # yesterday is the most recent day it can be updated through)
-          where(%{(document->>'pageview_count_updated_through') IS NULL \
-                OR ("canonical" = ? AND document->>'pageview_count_updated_through' < ?)},
-                true, 1.day.ago.to_date)
-            .where('host IS NULL OR host = ?', [ENV['DEFAULT_HOST']]) # Only query GA for paths on the current host
+        def pageview_counts
+          super || self.pageview_counts = {}
```

Comment:
have you checked out the attribute api? Does that work here, to initialize it to a hash?

Otherwise, a bit more straightforward way to do this would be in `initialize`

```
def initialize
  self.pageview_counts ||= { }
end
```